cache/network/download.go

```diff
@@ -45,31 +47,18 @@ func Download(params DownloadParams, logger log.Logger) (matchedKey string, err
 	}
 
 	logger.Debugf("Download archive")
-	file, err := os.Create(params.DownloadPath)
-	if err != nil {
-		return "", fmt.Errorf("can't open download location: %w", err)
-	}
-	defer func(file *os.File) {
-		err := file.Close()
-		if err != nil {
-			logger.Warnf(err.Error())
-		}
-	}(file)
 
-	respBody, err := client.downloadArchive(restoreResponse.URL)
-	if err != nil {
-		return "", fmt.Errorf("failed to download archive: %w", err)
-	}
-	defer func(respBody io.ReadCloser) {
-		err := respBody.Close()
-		if err != nil {
-			logger.Warnf(err.Error())
-		}
-	}(respBody)
-	_, err = io.Copy(file, respBody)
-	if err != nil {
-		return "", fmt.Errorf("failed to save archive to disk: %w", err)
+	downloadErr := downloadFile(retryableHTTPClient.StandardClient(), restoreResponse.URL, params.DownloadPath)
+	if downloadErr != nil {
+		return "", fmt.Errorf("failed to download archive: %w", downloadErr)
 	}
 
 	return restoreResponse.MatchedKey, nil
 }
+
+func downloadFile(client *http.Client, url string, dest string) error {
+	downloader := got.New()
```

Comment:
Does this lib have any kind of debug logging feature? We have a debug log param in the steps and would be great to support that

Reply:
As we discussed, we go with the current debug logs (we log with the underlying HTTP client)